src/Config.php

```diff
@@ -237,7 +248,12 @@ private function merge(array $context, string $path = '', array ...$args): array
                         /** @var mixed */
                         $result[$k] = $v;
                     }
-                } elseif (is_array($v) && isset($result[$k]) && is_array($result[$k])) {
+                } elseif (
+                    in_array($context[0], $this->mergeRecursive, true) &&
```

Comment:
`in_array` check is O(n). If we'll reverse `$mergeRecursive` in constructor, i.e. will store it as:

```php
[
    'params.php' => 1,
    'events.php' => 1,
]
```

then we'll be able to use `array_key_exists()` or `isset()` and that would be O(1).

Reply:
It is not a large array, you don't boost it much by this way. But you can do it if you think it's critical.